File: primatis-api/src/main/java/be/primatis/user/web/MeProfileResponse.java
package be.primatis.user.web;

import be.primatis.user.AccountStatus;
import be.primatis.user.AppUser;
import be.primatis.user.MemberStatus;

import java.time.LocalDate;

/**
 * Contrat REST du profil personnel de l'utilisateur authentifié (DEV-05.9,
 * {@code GET}/{@code PATCH /api/v1/me/profile}). Exclut volontairement
 * {@code roles}, {@code permissions} (déjà disponibles côté client via le
 * JWT, DEV-05.9 audit §5/§9 — ne pas dupliquer), {@code createdAt}/{@code
 * updatedAt}/{@code lastLoginAt} (aucun besoin identifié pour un profil
 * personnel) et {@code Residence} (DEV-05.9-DEC-01, reste sur {@code
 * /api/v1/me/residence}). Ne réutilise pas {@link
 * be.primatis.user.web.UserResponse} : contrat distinct, volontairement pas
 * automatiquement synchronisé avec la vue staff.
 */
public record MeProfileResponse(
        Long id,
        String email,
        String firstName,
        String lastName,
        String phoneNumber,
        AccountStatus accountStatus,
        String memberNumber,
        MemberStatus memberStatus,
        LocalDate registrationDate,
        LocalDate memberExpirationDate,
        String blockedReason) {

    public static MeProfileResponse from(AppUser appUser) {
        return new MeProfileResponse(
                appUser.getId(),
                appUser.getEmail(),
                appUser.getFirstName(),
                appUser.getLastName(),
                appUser.getPhoneNumber(),
                appUser.getAccountStatus(),
                appUser.getMemberNumber(),
                appUser.getMemberStatus(),
                appUser.getRegistrationDate(),
                appUser.getMemberExpirationDate(),
                appUser.getBlockedReason());
    }
}

File: primatis-web/src/app/user/models/me-profile-response.ts
import { AccountStatus } from './account-status';
import { MemberStatus } from './member-status';

/**
 * Voir `be.primatis.user.web.MeProfileResponse` côté backend. Exclut
 * volontairement `roles`/`permissions` (déjà disponibles côté frontend via
 * le JWT, ne pas dupliquer), `createdAt`/`updatedAt`/`lastLoginAt` et
 * `Residence` (reste sur `/api/v1/me/residence`, jamais imbriquée ici).
 */
export interface MeProfileResponse {
  id: number;
  email: string;
  firstName: string;
  lastName: string;
  phoneNumber: string | null;
  accountStatus: AccountStatus;
  memberNumber: string | null;
  memberStatus: MemberStatus | null;
  registrationDate: string | null;
  memberExpirationDate: string | null;
  blockedReason: string | null;
}
